packages/webcomponentsjs/ts_src/platform/flag-parser.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be found
+ * at http://polymer.github.io/AUTHORS.txt The complete set of contributors may
+ * be found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by
+ * Google as part of the polymer project is also subject to an additional IP
+ * rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+export {};
+
+interface ExtendedWindow extends Window {
+  WebComponents?: {flags?: Flags;};
+  ShadyDOM?: {force?: boolean|string; noPatch?: boolean | string;};
+  customElements: CustomElementRegistry&{
+    forcePolyfill?: string|boolean;
+  };
+}
```

Comment:
Why this vs `declare namespace window { ... }`?

Reply:
`declare namespace window { ... }` makes a global change for all code in the entire compilation unit (which could be a lot of code in a big app that imports this).

This style keeps the changes isolated.